packages/graphiql/cypress/integration/lint.spec.ts

```diff
@@ -0,0 +1,156 @@
+describe('Linting', () => {
+  it('Marks GraphQL syntax errors as error', () => {
+    cy.visitWithOp({
+      query: /* GraphQL */ `
+        {
+          doesNotExist
+          test {
+            id
+          }
+          +++
+        }
+      `,
+    }).assertLinterMarkWithMessage(
+      '+++',
+      'error',
+      'Syntax Error: Unexpected character: "+".',
```

Comment:
here you could use a version flag using `graphql/version`, and find the difference between these strings. annoying but it allows us to provide some backwards compatible support for 15

https://github.com/thomasheyenbrock/graphiql/blob/6efac7e5a565a13f9c3b2069098f51dbc3adfb4d/packages/graphiql/cypress/integration/incrementalDelivery.spec.ts#L6

first,

`import { version as graphqlVersion } from 'graphql/version';`

This is the only difference between major versions if i recall - the new syntax error ends with a `.`. there is probably a more terse way of doing this of course
```ts

// 
const syntaxError = graphqlVersion.includes('15') ? 'Syntax Error: Unexpected character: "+"' : 'Syntax Error: Unexpected character: "+".'

cy.visitWithOp({
      query: /* GraphQL */ `
        {
          doesNotExist
          test {
            id
          }
          +++
        }
      `,
    }).assertLinterMarkWithMessage(
      '+++',
      'error',
      syntaxError,
      )
 ```

Reply:
Thanks for the hint! Used a ternary based on the GraphQL version to resolve this as you proposed 👍